### PR TITLE
Add 404 page to website

### PR DESCRIPTION
**Pull request by ElianCodes**
## Changes

- Add a 404 page to the main website
- [maybe something cool for the future?](https://www.npmjs.com/package/@elianvancutsem/mostvisitedpages)

## Testing

- No tests added

## Docs

- No new documentation added, willing to write if needed

### CODE REVIEW

**Comment by jasikpark** [comment]
Looks good! Thanks for the PR! There are a couple of problems though? 

<details>
<summary>Homepage with White background and non-visible-space</summary>

image

</details>


<details>
<summary>404 page might need a "height: 100vh" or something, because it's scrunched down on mobile</summary>

image

</details>

File: www/src/pages/404.astro [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import MainHeader from '../components/MainHeader.astro';
+import Space from '../components/Space.astro';
+import ErrorPage from '../components/ErrorPage.astro';
+
+let title = 'Astro | Page Not Found';
+let description = 'Build faster websites with less client-side JavaScript';
+let permalink = 'https://astro.build/404';
+let lang = 'en';
+---
+<html lang={ lang ?? 'en' }>
```
Comment (aFuzzyBear) [inline www/src/pages/404.astro:13]: Is this necessary? I was just looking, the `lang` is already being hardcoded in the frontmatter, could you not just set the attribute, `<html lang='en'/>`

Comment (ElianVanCutsem) [reply www/src/pages/404.astro:13]: Yeah, noticed that too, but it's done in every page, so I decided to follow. I also noticed that in `<BaseHead />` the `permalink` property should be renamed to `canonicalURL`

**Comment by ElianVanCutsem** [comment]
Did some further research; The same issue comes up when adding any new page. So I'm guessing somewhere an element is not closed correctly or something. Any ideas?

**Pull request merged**